src/ObjLoading/Game/IW4/AssetLoaders/AssetLoaderLeaderboard.cpp

```diff
@@ -15,3 +18,31 @@ void* AssetLoaderLeaderboard::CreateEmptyAsset(const std::string& assetName, Mem
     leaderboard->name = memory->Dup(assetName.c_str());
     return leaderboard;
 }
+
+bool AssetLoaderLeaderboard::CanLoadFromRaw() const
+{
+    return true;
+}
+
+std::string AssetLoaderLeaderboard::GetFileNameForAsset(const std::string& assetName)
+{
+    return std::format("leaderboards/{}.json", assetName);
+}
```

Comment:
There are some things that are done in some places in the codebase that i would not do anymore that way:
This is one thing: I currently would not put these kinds of methods in the AssetLoaderLeaderboard class.

This is because this requires this method to be declared in the header, even though it is not used outside of the cpp file at all.
My current approach is to define these kinds of methods in an anonymous namespace in the cpp file (anonymous namespace because this avoids conflicts with other methods with that same name when linking).

In this case it would probably also make sense to omit the method entirely because its only used in one place and its a one liner, so you dont really reduce complexity by introducing a method.
The "GetFileNameForAsset" methods were mainly introduced to extract code when generating the filenames requires more logic.

All in all a lot of the stuff im going to comment is not anything tragic at all but i thought i'd just clearify my thoughts on these places since im currently doing it differently for new code :P